lib/manager/nuget/extract.ts

```diff
@@ -74,7 +74,7 @@ async function determineRegistryUrls(
   return registryUrls;
 }
 
-const packageRe = /<(?:PackageReference|DotNetCliToolReference).*Include\s*=\s*"([^"]+)".*Version\s*=\s*"(?:[[])?(?:([^"(,[\]]+)\s*(?:,\s*[)\]]|])?)"/;
+const packageRe = /<(?:PackageReference|DotNetCliToolReference|GlobalPackageReference).*(?:Include|Update)\s*=\s*"([^"]+)".*Version\s*=\s*"(?:[[])?(?:([^"(,[\]]+)\s*(?:,\s*[)\]]|])?)"/;
```

Comment:
This regex should be refactored to use regex groups for better readability and easy extension

Reply:
I'll admit, regex isn't my strong suit so I'm not quite sure what you mean, it's already using the match groups further down in the code based on this regex.  Is there an example somewhere else in the code of what you'd like that refactor to look like?

Reply:
ok, i'll update the regex later